src/app/features/auth/components/register-form/register-form.component.html

```diff
@@ -0,0 +1,115 @@
+<div class="relative">
+  <spinner *ngIf="loading" />
+  <form
+    class="bg-white shadow-md rounded px-8 pt-6 pb-8 mb-4 form-middle"
+    [formGroup]="form"
+    (ngSubmit)="registerUser()"
+  >
+    <ng-content></ng-content>
+    <div class="mb-4">
+      <label class="block text-gray-700 text-sm font-bold mb-2" for="name">
+        Name
+      </label>
+      <input
+        class="shadow appearance-none border rounded w-full py-2 px-3 text-gray-700 leading-tight focus:outline-none focus:shadow-outline"
+        formControlName="name"
+        id="name"
+        placeholder="Name"
+        type="text"
+      />
+      <div
+        *ngIf="
+          form.controls['name'].touched &&
+          form.controls['name']?.errors?.['required']
+        "
+        class="text-red-500 text-sm mt-1"
+      >
+        {{ formErrors['requiredName'] }}
```

Comment:
You can make you check a lot shorter if you name keys in the `formErrors` map the same way as angular error names:
```ts
export const formErrors: {
    required: 'This field is required',
    min: '...',
    max: '...'
}
```
This way you will just check if the field is invalid and was touched and pass whatever the error is there in the field as the key to your map.